### PR TITLE
opae: fix metric max10 value limits check

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

File: libopae/plugins/xfpga/metrics/metrics_int.h [hunk 1]
```diff
@@ -59,6 +59,14 @@
 #define VOLTAGE                             "Voltage"
 #define CURRENT                             "Current"
 #define POWER                               "Power"
+#define CLCOK                               "Clock"
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_int.h]: s/CLCOK/CLOCK/

File: libopae/plugins/xfpga/metrics/metrics_int.h [hunk 2]
```diff
@@ -59,6 +59,14 @@
...
+
+
+#define TEMPERATURE_UNITS                   "Celsius"
+#define VOLTAGE_UNITS                       "Volts"
+#define CURRENT_UNITS                       "Amps"
+#define POWER_UNITS                         "Watts"
+#define CLCOK_UNITS                         "Hz"
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_int.h]: s/CLCOK_/CLOCK_/

File: libopae/plugins/xfpga/metrics/metrics_max10.c [hunk 3]
```diff
@@ -59,6 +59,14 @@
 #include "metrics/metrics_metadata.h"
 #include "metrics/max10_metadata.h"
 
+// Max10 Metric limits
+#define THERMAL_HIGH_LIMIT             300.00
+#define THERMAL_LOW_LIMIT              -273.00
+#define POWER_HIGH_LIMIT               1000.00
+#define POWER_LOW_LIMIT                0.00
+#define VOLTAMP_HIGH_LIMIT             500.00
+#define VOLTAMP_LOW_LIMIT              0.00
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_max10.c]: These limits are true for all MAX10, regardless of the platform?

File: libopae/plugins/xfpga/metrics/metrics_max10.c [hunk 4]
```diff
@@ -258,23 +266,68 @@ fpga_result  enum_max10_metrics_info(struct _fpga_handle *_handle,
...
 		}
 
+		// Metric Units
+		if (EOK == strstr_s(metric_name, strlen(metric_name),
+				POWER, strlen(POWER), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				POWER_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_max10.c]: Just continuing here silently ignores the error.

File: libopae/plugins/xfpga/metrics/metrics_max10.c [hunk 5]
```diff
@@ -258,23 +266,68 @@ fpga_result  enum_max10_metrics_info(struct _fpga_handle *_handle,
...
+				}
+
+		} else if (EOK == strstr_s(metric_name, strlen(metric_name),
+				VOLTAGE, strlen(VOLTAGE), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				VOLTAGE_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_max10.c]: Just continuing here silently ignores the error.

File: libopae/plugins/xfpga/metrics/metrics_max10.c [hunk 6]
```diff
@@ -258,23 +266,68 @@ fpga_result  enum_max10_metrics_info(struct _fpga_handle *_handle,
...
+				}
+
+		} else if (EOK == strstr_s(metric_name, strlen(metric_name),
+					CURRENT, strlen(CURRENT), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				CURRENT_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_max10.c]: Just continuing here silently ignores the error.

File: libopae/plugins/xfpga/metrics/metrics_max10.c [hunk 7]
```diff
@@ -258,23 +266,68 @@ fpga_result  enum_max10_metrics_info(struct _fpga_handle *_handle,
...
+				}
+
+		} else if (EOK == strstr_s(metric_name, strlen(metric_name),
+				TEMPERATURE, strlen(TEMPERATURE), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				TEMPERATURE_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_max10.c]: Just continuing here silently ignores the error.

File: libopae/plugins/xfpga/metrics/metrics_max10.c [hunk 8]
```diff
@@ -258,23 +266,68 @@ fpga_result  enum_max10_metrics_info(struct _fpga_handle *_handle,
...
+				}
+
+		} else if (EOK == strstr_s(metric_name, strlen(metric_name),
+					CLCOK, strlen(CLCOK), &substr)) {
+
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				CLCOK_UNITS, SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_max10.c]: Just continuing here silently ignores the error.

File: libopae/plugins/xfpga/metrics/metrics_max10.c [hunk 9]
```diff
@@ -258,23 +266,68 @@ fpga_result  enum_max10_metrics_info(struct _fpga_handle *_handle,
...
+				}
+
+		} else {
+			e = strncpy_s(metric_units, sizeof(metric_units),
+				"N/A", SYSFS_PATH_MAX);
+				if (EOK != e) {
+				continue;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics_max10.c]: Just continuing here silently ignores the error.

File: libopae/plugins/xfpga/metrics/metrics_max10.c [hunk 10]
```diff
@@ -287,5 +340,66 @@ fpga_result  enum_max10_metrics_info(struct _fpga_handle *_handle,
...
+fpga_result read_max10_value(struct _fpga_enum_metric *_fpga_enum_metric,
+					double *dvalue)
+{
+	fpga_result result     = FPGA_OK;
+	char *substr           = NULL;
+	uint64_t value         = 0;
+
+	if (_fpga_enum_metric == NULL ||
+		dvalue == NULL) {
+		FPGA_ERR("Invalid Input Paramters");
```
Comment (ibarraza) [inline libopae/plugins/xfpga/metrics/metrics_max10.c]: parameters

File: libopae/plugins/xfpga/metrics/metrics_int.h (hunk 1 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/metrics/metrics_int.h]: fixed

Comment (anandaravuri) [reply libopae/plugins/xfpga/metrics/metrics_int.h]: fixed

File: libopae/plugins/xfpga/metrics/metrics_int.h (hunk 2 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/metrics/metrics_int.h]: fixed

File: tools/base/fpgainfo/board.c [hunk 11]
```diff
@@ -269,7 +269,10 @@ fpga_result mac_command(fpga_token *tokens, int num_tokens, int argc,
 
 		fpgainfo_board_info(tokens[i]);
 		fpgainfo_print_common("//****** MAC ******//", props);
-		mac_info(tokens[i]);
+		res = mac_info(tokens[i]);
+		if (res != FPGA_OK) {
+			printf("mac info is not supported\n");
```
Comment (tswhison) [inline tools/base/fpgainfo/board.c]: Use OPAE_ERR

File: tools/base/fpgainfo/board.c [hunk 12]
```diff
@@ -386,7 +389,10 @@ fpga_result phy_command(fpga_token *tokens, int num_tokens, int argc,
 
 		fpgainfo_board_info(tokens[i]);
 		fpgainfo_print_common("//****** PHY ******//", props);
-		phy_group_info(tokens[i]);
+		res = phy_group_info(tokens[i]);
+		if (res != FPGA_OK) {
+			printf("phy group info is not supported\n");
```
Comment (tswhison) [inline tools/base/fpgainfo/board.c]: Use OPAE_ERR

File: libopae/plugins/xfpga/metrics/metrics_max10.c (hunk 3 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/metrics/metrics_max10.c]: yes
Max10 /BMC limits

File: libopae/plugins/xfpga/metrics/metrics_max10.c (hunk 10 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/metrics/metrics_max10.c]: fixed

File: tools/base/fpgainfo/board.c [hunk 13]
```diff
@@ -413,7 +419,7 @@ fpga_result fpgainfo_board_info(fpga_token token)
 	if (print_board_info) {
 		res = print_board_info(token);
 	} else {
-		OPAE_ERR("No print_board_info entry point:%s\n", dlerror());
+		OPAE_MSG("No print_board_info entry point:%s\n", dlerror());
```
Comment (tswhison) [inline tools/base/fpgainfo/board.c]: I thought OPAE_ERR was appropriate for these three, since you're returning an error code.

Comment (anandaravuri) [reply tools/base/fpgainfo/board.c]: Some platforms like MCP doesn't support board info, fpgainfo out put shows error message.

File: tools/base/fpgainfo/board.c (hunk 12 above)
Comment (anandaravuri) [reply tools/base/fpgainfo/board.c]: RC platform doesn't support MAC and PHY commends.  Prints not supported if user pass comments to fpgainfo .

**Pull request merged**